### DISCUSSION TITLE
ISR: how to trigger revalidation if the same user comes over the different user-agent

### DISCUSSION THREAD

**Discussion by oyilmaztekin** [vercel/next.js#29096] [Help] [answered]
The application has the rule to serve pages from the SSR based on the device that users use. The device detection logic relies on the user-agent and only occurs on the server without changing the URL structure. The way that the app use is by serving the page over the same URL regardless of devices. Unfortunately, if the user came from the mobile browser for the first visit and the subsequent requests come from the desktop or tablet device, the cache serves the response created for the mobile user. 

I know it's a little bit tricky, but is there a way to add a condition inside of the  `getStaticProps` to trigger revalidation something like this : 

```js
export async function getStaticProps({ params, req}) {
// prevUserAgent ???
  return {
    props: {
      product: await getProductFromDatabase(params.id),
    },
    revalidate: prevUserAgent !== req?.headers['user-agent'],
  };
}
```

**Comment by wilsonneto-swe** [comment, marked as answer]
Hi @oyilmaztekin , I think what you want isn't possible and isn't the getStaticProps motivation... The getStaticProps will answer the same cache even when another user in another country with another browser/user-agent access the site.

If you need a cache by user-agente, will be better to deal with this on frontend having different URL for different user-agents, because the way you said, if a opera user access your site he will clear all the cache even for chrome users... I don't know if I understood weel...

You can do something like, but I don't recommend, with custom server:
https://nextjs.org/docs/advanced-features/custom-server

I hope it helps you.
Best regards.
